### ISSUE TITLE
haxe.macro.ClassType#constructor inconsistency

### ISSUE THREAD

**Issue by andyli** [HaxeFoundation/haxe#3738]
``` haxe
import haxe.macro.*;
import haxe.macro.Type;

class A {
    public function new():Void {}
}

class AA extends A {

}

class Test {
    macro static public function check():Expr {
        switch (Context.getType("AA")) {
            case TInst(t, _):
                trace(t.get().constructor == null); //false for all targets except Java/C#
            case _:
        }
        return macro {};
    }
    static function main():Void {
        check();
    }
}
```

**Comment by Simn** [comment]
@waneck: What are the plans for `add_constructor` on Java and Cs? Are we going to deal with overloads there at some point?

**Comment by Simn** [comment]
The Java and C# targets will be put in "regression fixes only" state with the Haxe 4.3.0 release. See https://github.com/HaxeFoundation/haxe/wiki/Target-Support-Level#java-and-c-since-430 for additional information. We apologize for any inconvenience this causes!
